### PR TITLE
WT-3971 Cursor duplicates use the cursor cache.

### PR DESCRIPTION
**Pull request by ddanderson**
This fixes a cursor cache leak of sorts when dups of the same URI are repeatedly made.

### CODE REVIEW

**Comment by ddanderson** [comment]
@sueloverso, if you're available, can you please review?

File: test/suite/test_cursor13.py [hunk 1]
```diff
@@ -538,6 +538,7 @@ def test_dup(self):
         cursor.close()
 
         c1 = self.session.open_cursor(uri, None)
+        c1.next()
```
Comment (sueloverso) [inline test/suite/test_cursor13.py]: Why do we need this line? If it is necessary please add a comment.

File: src/cursor/cur_std.c [hunk 2]
```diff
@@ -751,11 +751,16 @@ __wt_cursor_cache_get(WT_SESSION_IMPL *session, const char *uri,
 			return (WT_NOTFOUND);
 	}
 
+	if (to_dup != NULL) {
+		uri = to_dup->uri;
+		hash_value = to_dup->uri_hash;
```
Comment (sueloverso) [inline src/cursor/cur_std.c]: I'd add a comment that we expect the caller to check that one of `uri` or `to_dup` is NULL.

File: test/suite/test_cursor13.py [hunk 3]
```diff
@@ -527,3 +527,22 @@ def test_cursor_sweep(self):
...
+    def test_dup(self):
+        self.cursor_stats_init()
+        uri = 'table:test_cursor13_dup'
+        self.session.create(uri, 'key_format=S,value_format=S')
+        cursor = self.session.open_cursor(uri)
+        cursor['A'] = 'B'
+        cursor.close()
+
+        c1 = self.session.open_cursor(uri, None)
+        c1.next()
```
Comment (sueloverso) [inline test/suite/test_cursor13.py]: It took a moment to understand the `next` call. Add a short comment that we need to position the cursor on the data before dup'ing it.

File: test/suite/test_cursor13.py [hunk 4]
```diff
@@ -527,3 +527,22 @@ def test_cursor_sweep(self):
...
+        for notused in range(0, 100):
+            self.session.breakpoint()
+            c2 = self.session.open_cursor(None, c1, None)
+            c2.close()
+        stats = self.caching_stats()
+        self.assertGreaterEqual(stats[0], 100)
```
Comment (sueloverso) [inline test/suite/test_cursor13.py]: A comment about what stats we're checking here would be helpful.

**Comment by agorrod** [comment]
lgtm

**Pull request merged**